### PR TITLE
Add workflows

### LINKED ISSUE #109: 🚧Add workflows
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Add sync and "add new" reusable workflows for better automation

### Acceptance Criteria

- [x] add new issue to project yml added
- [x] all sync yml added

### ToDo Items

- [X] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [X] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [X] Issue linked to the correct project _(if applicable)_.
- [X] Issue linked to the correct milestone _(if applicable)_.
- [x] Draft pull request created and linked to this issue _(only required with code changes)_.

### Issue Dependencies

_No response_

### Related Work

_No response_

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                |
|---------------------|----------------------|
| Bug Fixes           | `🐛bug` |
| Breaking Changes    | `🧨breaking changes` |
| New Feature         | `✨new feature` |
| CICD Changes        | `♻️cicd` |
| Code Doc Changes    | `🗒️documentation/code` |
| Product Doc Changes | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### PR DESCRIPTION
**Pull request by kselena**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">📋Review Checklist📋</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/feature/<issue-num>-<description>
      Example: preview/feature/123-my-preview-feature
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_preview release_** branch, with the correct naming syntax, in the following example:
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
    Example: preview/v1.2.3-preview.4
    ```
  </details>

- [x] The **_base(target)_** branch for this pull request is a **_preview release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

💡For more information on branching, refer to the project's [branching documentation](https://github.com/KinsonDigital/.github/blob/master/docs/BranchingFullGitFlow.md).

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title matches the title of the associated issue.

- [x] A `preview` label has been added to the pull request.

- [x] The labels attached to this PR match the labels attached to the associated issue.

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out, and let's discuss it.

- [x] I have manually tested my code changes to the best of my ability.

- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - Status checks run for this pull request when merging into the **_preview/v\*.\*.\*-preview.\*_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

File: .github/workflows/add-new-issue-to-project.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+jobs:
+  add_new_issue_to_project:
+    name: Add New Issue
+    uses: KinsonDigital/Infrastructure/.github/workflows/add-issue-to-project.yml@v8.2.0
+    with:
+      org-name: "${{ vars.ORGANIZATION_NAME }}"
+      org-project-name: "${{ vars.ORG_PROJECT_NAME }}"
+      project-name: "${{ vars.PROJECT_NAME }}"
+    secrets:
+      cicd-pat: ${{ secrets.CICD_REST_API }}
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/add-new-issue-to-project.yml:18]: Update token reference:

```suggestion
      cicd-pat: ${{ secrets.CICD_ TOKEN }}
```

File: .github/workflows/sync-bot.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            3. Event Type - set to issue event type
+            4. GitHub token
+          #>
+          deno run `
+            --allow-net `
+            "$scriptUrl" `
+            "${{ vars.PROJECT_NAME }}" `
+            "$issueNumber" `
+            "issue" `
+            "${{ secrets.CICD_REST_API }}";
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/sync-bot.yml:51]: Update token reference:

```suggestion
      cicd-pat: ${{ secrets.CICD_ TOKEN }}
```

File: .github/workflows/sync-issue-to-pr.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+          deno run `
+            --allow-net `
+            "$scriptUrl" `
+            "${{ vars.ORGANIZATION_NAME }}" `
+            "${{ vars.PROJECT_NAME }}" `
+            "${{ github.event.sender.login }}" `
+            "$prNumber" `
+            "$command" `
+            "${{ secrets.CICD_REST_API }}";
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/sync-issue-to-pr.yml:51]: Update token reference:

```suggestion
            "${{ secrets.CICD_ TOKEN }}";
```

File: .github/workflows/sync-status-check.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            3. Event Type - set to pull request event type
+            4. GitHub token
+          #>
+          deno run `
+            --allow-net `
+            "$scriptUrl" `
+            "${{ vars.PROJECT_NAME }}" `
+            "$prNumber" `
+            "pr" `
+            "${{ secrets.CICD_REST_API }}";
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/sync-status-check.yml:51]: Update token reference:

```suggestion
            "${{ secrets.CICD_TOKEN }}";
```

File: .github/workflows/add-new-issue-to-project.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+on:
+  issues:
+    types: opened
+
+
+jobs:
+  add_new_issue_to_project:
+    name: Add New Issue
+    uses: KinsonDigital/Infrastructure/.github/workflows/add-issue-to-project.yml@v8.2.0
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/add-new-issue-to-project.yml:12]: 1. Update version:

```suggestion
    uses: KinsonDigital/Infrastructure/.github/workflows/add-issue-to-project.yml@v9.0.0
```

2. Update the repository variable named `CICD_SCRIPTS_VERSION` to the same version

**Review by KinsonDigitalAdmin — DISMISSED** [review]
M

**Review by KinsonDigitalAdmin — APPROVED** [review]
